src/components/ha-expansion-panel.ts

```diff
@@ -0,0 +1,141 @@
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import { fireEvent } from "../common/dom/fire_event";
+import "./ha-svg-icon";
+import { mdiChevronDown } from "@mdi/js";
+import { classMap } from "lit-html/directives/class-map";
+
+@customElement("ha-expansion-panel")
+class HaExpansionPanel extends LitElement {
+  @property({ type: Boolean, reflect: true }) expanded = false;
+
+  @property({ type: Boolean, reflect: true }) outlined = false;
+
+  protected render(): TemplateResult {
+    return html`
+      <div class="summary" @click=${this._toggleContainer}>
+        <slot name="title"></slot>
+        <ha-svg-icon
+          .path=${mdiChevronDown}
+          class="summary-icon ${classMap({ expanded: this.expanded })}"
+        ></ha-svg-icon>
+      </div>
+      <div class="container ${classMap({ expanded: this.expanded })}">
+        <slot></slot>
+      </div>
+    `;
+  }
+
+  protected firstUpdated(changedProps) {
+    super.firstUpdated(changedProps);
+
+    const container: HTMLDivElement = this.shadowRoot!.querySelector(
+      ".container"
+    )! as HTMLDivElement;
+
+    if (this.expanded) {
+      setTimeout(() => {
+        const scrollHeight = container.scrollHeight;
+        const animationDuration = this.getAutoHeightDuration(scrollHeight);
+        container.style.height = `${scrollHeight}px`;
+        container.style.transitionDuration = `${animationDuration}ms`;
+      }, 0);
+    }
+  }
+
+  private _toggleContainer(): void {
+    const container: HTMLDivElement = this.shadowRoot!.querySelector(
+      ".container"
+    )! as HTMLDivElement;
+    this.expanded = !this.expanded;
+
+    if (this.expanded) {
+      const scrollHeight = container.scrollHeight;
+      const animationDuration = this.getAutoHeightDuration(scrollHeight);
+      container.style.height = `${scrollHeight}px`;
```

Comment:
You should remove the height when the transition is done.

Reply:
Ok. Is there a helper I can use or should I use:
```
x.addEventListener("animationend", myEndFunction);
```

Reply:
`@animationend=${this._handleAnimationEnd}`

Reply:
done with `@transitionend`